tests/unit/test_async_util.py

```diff
@@ -238,3 +240,20 @@ async def test_asyncqgen():
         ret.append(item)
 
     assert ret == [1, 2, 3]
+
+
+async def test_scandir(tmp_path: Path):
+    """It should list directory contents (including symlinks)."""
+    (tmp_path / 'a').touch()
+    (tmp_path / 'b').touch()
+    (tmp_path / 'c').symlink_to(tmp_path / 'b')
+
+    assert sorted(await scandir(tmp_path)) == [
+        Path(tmp_path, 'a'),
+        Path(tmp_path, 'b'),
+        Path(tmp_path, 'c')
+    ]
+
+    # Should not raise if the given path doesn't exist. This matches the
+    # previous pyuv implementation.
+    assert await scandir(tmp_path / 'HORSE') == []
```

Comment:
Not willing to go too far with this one, but IMO smaller tests make debugging easier, esp when there's no reason to put them in a single place.

```suggestion
async def test_scandir_non_exist(tmp_path: Path):
    """It returns empty list if target dir doesn't exist

    This matches the previous pyuv implementation.
    """
    assert await scandir(tmp_path / 'HORSE') == []
```